File: back/src/main/java/com/models/TournamentUser/TournamentUser.java
package com.models.TournamentUser;

import com.models.Tournament.Tournament;
import com.models.User.User;
import lombok.*;

import javax.persistence.*;

@Data
@Entity
@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
public class TournamentUser {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "id", nullable = false, unique = true)
    private Long id;
    @ManyToOne(targetEntity = Tournament.class, fetch = FetchType.EAGER, cascade = CascadeType.ALL)
    @JoinColumn(name = "tournament.id")
    private Tournament tournament_id;
    @ManyToOne(targetEntity = User.class, fetch = FetchType.EAGER, cascade = CascadeType.ALL)
    @JoinColumn(name = "user.id")
    private User user_id;
    private String birth_certificate;
    private String pay_certificate;
    private Boolean is_confirmed;
}

File: front/src/app/Models/TournamentUser.ts
import {Tournament} from './Tournament';
import {User} from './User';

export class TournamentUser {
    id: number;
    tournament_id: Tournament;
    user_id: User;
    birth_certificate: string;
    pay_certificate: string;
    is_confirmed: boolean;

  constructor(birth_certificate: string, pay_certificate: string) {
    this.birth_certificate = birth_certificate;
    this.pay_certificate = pay_certificate;
  }
}
